homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,274 @@
+        return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature for the thermostat."""
+        return 30
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        return self.send({self._api.TEMP: temp})
+
+    def set_fan_mode(self, fan):
+        """Set fan mode."""
+        fan_mode = self.hass_fan_to_melissa(fan)
+        return self.send({self._api.FAN: fan_mode})
+
+    def set_operation_mode(self, operation_mode):
+        """Set operation mode."""
+        mode = self.hass_mode_to_melissa(operation_mode)
+        return self.send({self._api.MODE: mode})
+
+    def turn_on(self):
+        """Turn on device."""
+        return self.send({self._api.STATE: self._api.STATE_ON})
+
+    def turn_off(self):
+        """Turn off device."""
+        return self.send({self._api.STATE: self._api.STATE_OFF})
+
+    def send(self, value):
+        """Sending action to service."""
+        try:
+            old_value = self._cur_settings.copy()
+            self._cur_settings.update(value)
+        except AttributeError:
+            old_value = None
+        if not self._api.send(self._serial_number, self._cur_settings):
+            self._cur_settings = old_value
+            return False
+        else:
+            return True
+
+    def update(self):
+        """Get latest data from Melissa."""
+        try:
+            self._data = self._api.status(cached=True)[self._serial_number]
+            self._cur_settings = self._api.cur_settings(
+                self._serial_number
+            )['controller']['_relation']['command_log']
+        except KeyError:
+            _LOGGER.warning(
+                'Unable to update component %s', self.entity_id)
+
+    def melissa_state_to_hass(self, state):
+        """Translate Melissa states to hass states."""
+        if state == self._api.STATE_ON:
+            return STATE_ON
+        elif state == self._api.STATE_OFF:
+            return STATE_OFF
+        elif state == self._api.STATE_IDLE:
+            return STATE_IDLE
+        else:
+            return STATE_UNKNOWN
+
+    def melissa_op_to_hass(self, mode):
+        """Translate Melissa modes to hass states."""
+        if mode == self._api.MODE_AUTO:
+            return STATE_AUTO
+        elif mode == self._api.MODE_HEAT:
+            return STATE_HEAT
+        elif mode == self._api.MODE_COOL:
+            return STATE_COOL
+        elif mode == self._api.MODE_DRY:
+            return STATE_DRY
+        elif mode == self._api.MODE_FAN:
+            return STATE_FAN_ONLY
+        else:
+            _LOGGER.warning(
+                "Operation mode %s could not be mapped to hass", mode)
+            return STATE_UNKNOWN
+
+    def melissa_fan_to_hass(self, fan):
```

Comment:
Replace this with a dict, mapping the home assistant fan state to the melissa fan modes. Put the dict on the module level.